secureli/abstractions/pre_commit.py

```diff
@@ -72,22 +74,39 @@ def __init__(
     ):
         self.command_timeout_seconds = command_timeout_seconds
 
-    def install(self, folder_path: Path):
+    def install(self, folder_path: Path) -> InstallResult:
         """
-        Creates the pre-commit hook file in the .git directory so that `secureli scan` is run on each commit
+        Creates a new  pre-commit hook file in .git/hooks with the SeCureLI pre-commit hook contents.
+        If a pre-commit hook file already exists, the existing file is copied as a back up and
+        overwritten with the SeCureLI hook contents.
         """
 
-        # Write pre-commit with invocation of `secureli scan`
-        pre_commit_hook = folder_path / ".git/hooks/pre-commit"
+        pre_commit_hook = Path(folder_path / ".git/hooks/pre-commit")
+        backup_hook_path: str = None
+
+        if pre_commit_hook.is_file():
+            timestamp = re.sub(r"[.:-]+", "", datetime.datetime.now().isoformat())
```

Comment:
I'd recommend adding a comment explaining what this is doing:
```suggestion
            # strip out certain chars from the ISO8601 timestamp for inclusion in file name
            timestamp = re.sub(r"[.:-]+", "", datetime.datetime.now().isoformat())
```